moveos/moveos-verifier/src/metadata.rs

```diff
@@ -340,18 +361,20 @@ fn is_defined_in_current_module(
                 view.identifier_at(shandle.name).to_string(),
             )
         }
-        SignatureToken::TypeParameter(_)
-        | SignatureToken::Bool
+        // Primitive types are allowed.
```

Comment:
Should we support Primitive type and Address?

Reply:
In the file moveos-stdlib/sources/bcd.move, there is the following code:

```move
module moveos_std::bcd{
    public fun to_bool(v: vector<u8>): bool {
        from_bytes<bool>(v)
    }

    public fun to_u8(v: vector<u8>): u8 {
        from_bytes<u8>(v)
    }

    public fun to_u64(v: vector<u8>): u64 {
        from_bytes<u64>(v)
    }

    public fun to_u128(v: vector<u8>): u128 {
        from_bytes<u128>(v)
    }

    public fun to_address(v: vector<u8>): address {
        from_bytes<address>(v)
    }

    #[private_generics(MoveValue)]
    /// Function to deserialize a type T.
    /// Note the `private_generics` ensure only the `MoveValue`'s owner module can call this function
    public native fun from_bytes<MoveValue>(bytes: vector<u8>): MoveValue;
}
```

Reply:
This is a specific case, we should not support Primitive type and Address in `private_generics`. We need to refactor the `bcd`:

1. Make the native function `from_bytes` private and without annotation.
2. Provide a public function and add `private_generics`, maybe `public_from_bytes` or `generic_from_bytes`.
3. the `to_xx` function calls the private `from_bytes`.

Reply:
Understood. Do we need to prepare to solve these problems?
If we are not ready to solve them yet, I will exclude the `bcd::from_bytes` function from the scope of verification in the verifiter for now.

Reply:
`1. Make the native function from_bytes private and without annotation`

To solve this problem, we need to handle it specially in the verifier. We need to implement the same functionality as #[private_generics] without requiring annotations.

Reply:
You can refactor it in this PR or just remove `private_generics` from `from_bytes` and record an issue.

Reply:
I have temporarily removed the `#[private_generics]` attribute from `from_bytes<T>` and recorded an issue at https://github.com/rooch-network/rooch/issues/210 and will handle this problem later.